src/main/java/com/google/step/snippet/external/StackOverflowClient.java

```diff
@@ -91,6 +84,9 @@ private String getQuestionId(String url) {
     }
     // Parse the URL to get the question id.
     String[] segments = uri.getPath().split("/");
+    if (segments.length <= ID_INDEX || !segments[QUESTION_INDEX].equals(QUESTION_PARAMETER)) {
```

Comment:
Also for `QUESTION_INDEX`, I feel like it might be better if it was named something like `QUESTION_PATH_INDEX` or something that better indicates what it represents? Not sure if you have a better name?